### PR TITLE
[minor] gc_ctrl.c:caml_gc_set cleanup

### PR DESCRIPTION
**Pull request by gasche**
This is a minor cleanup which I suggested in https://github.com/ocaml/ocaml/pull/11713#discussion_r1030437311 but did not make the cut in the final version of that PR.

The idea is to make it clear which fields of the structure are used, in a way that obviously matches `caml_gc_get`.

https://github.com/gasche/ocaml/blob/be2db8e1362dfdd930544f10698859b1ad322237/runtime/gc_ctrl.c#L129-L135

cc @Octachron @Engil

### CODE REVIEW

**Comment by Octachron** [comment]
Personally I like the change: reading the ocaml record at one location in the C code seems clearer.

File: runtime/gc_ctrl.c [hunk 1]
```diff
@@ -156,40 +156,40 @@ static uintnat norm_custom_min (uintnat p)
 
 CAMLprim value caml_gc_set(value v)
 {
-  uintnat newpf;
-  uintnat newminwsz;
-  uintnat new_custom_maj, new_custom_min, new_custom_sz;
-  CAML_EV_BEGIN(EV_EXPLICIT_GC_SET);
+  uintnat newminwsz = caml_norm_minor_heap_size (Long_val (Field (v, 0)));
+  uintnat newpf = norm_pfree (Long_val (Field (v, 2)));
```
Comment (Octachron) [inline runtime/gc_ctrl.c:160]: Naive question: why was the documentation for the `major_heap_increment` not documented to reflect the fact that the parameter is not currently used anywhere in the runtime?

Comment (gasche) [reply runtime/gc_ctrl.c:160]: I have no idea. My guess: it's work and no one has done it yet. Note that there are many things that would require more work in the GC statistics (we have to rework the system to understand which stats to make domain-local and which ones to report globally, the computation of the maxima is wrong, etc.), so the best thing we can say for 5.0 is that it is a Work in Progress.

**Comment by gasche** [comment]
@Octachron if you believe that the change is correct, I would be interested in an approval.

**Review by abbysmal — APPROVED** [review]
Looks good to me, and it makes things much clearer.
I think we can merge this.

I also agree that we should get back to the documentation and discussions are the semantics of gc stats in the Multicore context: both are pretty intertwined, and should be discussed at the same time.

Thanks for your PR!

**Comment by gasche** [comment]
Thanks!

It looks like the `macos` CI configuration is broken right now, so I'm ignoring its output and merging.

**Pull request merged**